docs/api-markdown-documenter/build-api-nav.js

```diff
@@ -0,0 +1,76 @@
+const { ApiItemKind, ApiItemUtilities } = require("@fluid-tools/api-markdown-documenter");
+
+const fs = require("fs-extra");
+const path = require("path");
+const yaml = require("js-yaml");
+
+/**
+ * Processes documents and generates data required for the nav bar.
+ * @param {Array<Object>} documents - List of documents with apiItem.
+ * @param {ApiItem | undefined} documents.apiItem - The API item that the document is created from. Some documents may not have an apiItem.
+ */
+async function buildNavBar(documents) {
+	const validKinds = new Set([
+		ApiItemKind.Class,
+		ApiItemKind.Interface,
+		ApiItemKind.Enum,
+		ApiItemKind.Namespace,
+	]);
+	const { allAPIs, packageMap } = documents.reduce(
+		({ allAPIs, packageMap }, { apiItem }) => {
+			if (apiItem === undefined) {
+				return { allAPIs, packageMap };
+			}
+
+			const { displayName, kind } = apiItem;
+
+			const associatedPackage = apiItem.getAssociatedPackage();
+			const packageName =
```

Comment:
We may just want to assert that `associatedPackage` isn't undefined. I don't believe we will hit that case if `apiItem` isn't undefined, and we rely on it being defined in both code-paths below.